src/test/java/com/fasterxml/jackson/databind/deser/PolymorphicIdClassDeserTest.java

```diff
@@ -0,0 +1,39 @@
+package com.fasterxml.jackson.databind.deser;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializeUsingJDKTest;
+import org.junit.jupiter.api.Test;
+
+import static com.fasterxml.jackson.databind.testutil.DatabindTestUtil.newJsonMapper;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class PolymorphicIdClassDeserTest {
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
+    @JsonSubTypes({@JsonSubTypes.Type(value = FooClassImpl.class)})
```

Comment:
As per earlier comments, this is unnecessary for `JsonTypeInfo.Id.CLASS` (won't be used).
(doesn't necessarily but may be confusing for anything reading)